src/components/util.tsx

```diff
@@ -243,6 +243,15 @@ const getNodeRIDsFromURL = (href) => {
   return decodedNodes;
 };
 
+/**
+ * wrapper to prevent issue with arrays such as [1, 's'] inferring their type as (string | number)[] instead of [number, string]
+ * while this can be accomplished using `as const` for simple cases such as the example, if the array contains an object, then
+ * using as const also makes that object and all its properties readonly
+ */
+function tuple<A1, A2, ARest>(a1: A1, a2: A2, ...rest: ARest[]): [A1, A2, ...ARest[]] {
```

Comment:
does this mean you can only define the first two types and then everything after that just gets the same type?